cidc_api/models/models.py

```diff
@@ -970,6 +972,61 @@ def file_purpose(self):
     def file_purpose(cls):
         return FILE_PURPOSE_CASE_CLAUSE
 
+    @property
+    def cimac_id(self):
+        """
+        Extract the `cimac_id` associated with this file, if any, by searching the file's 
+        additional metadata for a field with a key like `<some>.<path>.cimac_id`.
+
+        NOTE: this is not a sqlalchemy hybrid_property, and it can't be used directly in queries.
+        """
+        for key, value in self.additional_metadata.items():
+            if key.endswith("cimac_id"):
+                return value
+        return None
+
+    @with_default_session
+    def get_related_files(self, session: Session) -> list:
+        """
+        Return a list of file records related to this file. We could define "related"
+        in any number of ways, but currently, a related file:
+            * is sample-specific, and relates to the same sample as this file if this file 
+              has an associated `cimac_id`.
+            * isn't sample-specific, and relates to the same `data_category_prefix`.
+        """
+        if self.cimac_id is not None:
+            query = text(
+                "SELECT downloadable_files.* "
+                "FROM downloadable_files, LATERAL jsonb_each_text(additional_metadata) addm_kv "
```

Comment:
I think this could potentially return multiple instances of the same DF record, in a case where `additional_metadata={"something.cimac_id": "Csomecimac.id", "something_else.cimac_id": "Csomecimac.id"}`
I'm mentioning this as I'm not sure if client code will be prepared to that.

Reply:
Would this be resolved by doing `SELECT DISTINCT downloadable_files.* ...` instead? I'll ensure the tests cover this case, either way, since it's a sneaky one.